index.js

```diff
@@ -168,7 +174,29 @@ class KeyringController extends EventEmitter {
    * @returns {Promise<Object>} A Promise that resolves to the state.
    */
   async submitPassword(password) {
+    await this.verifyPassword(password);
     this.keyrings = await this.unlockKeyrings(password);
+
+    // If we're provided a password, we should persist keyrings
```

Comment:
Nit: this comment should be updated because a password is not optional at this point. I might reword this as: `persistAllKeyrings` is here called so that we can either (1) migrate or (2) create a new salt